robot_ws/src/person_detection_robot/launch/kinesis.launch.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+import os
+import yaml
+
+from ament_index_python.packages import get_package_share_directory
+
+import launch
+from launch.substitutions import PythonExpression
+from launch.conditions import IfCondition
+from launch.substitutions import LaunchConfiguration
+
+def generate_launch_description():
+    kvs_config_file_path = os.path.join(
+        get_package_share_directory('person_detection_robot'),
+        'config', 'kvs_config.yaml')
+    h264_encoder_config_file_path = os.path.join(
+        get_package_share_directory('person_detection_robot'),
+        'config', 'h264_encoder_config.yaml')
+    kvs_logger_config_file_path = os.path.join(
+        get_package_share_directory('kinesis_video_streamer'),
+        'config', 'kvs_log_configuration')
+
+    with open(kvs_config_file_path, 'r') as f:
+        config_text = f.read()
+    config_yaml = yaml.safe_load(config_text)
+
+    default_stream_name = config_yaml['kinesis_video_streamer']['ros__parameters']['kinesis_video']['stream0']['stream_name']
+
+    launch_actions = [
+        launch.actions.DeclareLaunchArgument(
+            name='launch_id',
+            description='Used for resource name suffix if specified',
```

Comment:
Which resource names are suffixed with the `LAUNCH_ID`? When I tried it, the Kinesis Video Stream name was exactly what was defined in `kvs_config.yaml` as `stream_name`, without any suffixes.

Reply:
Yes, launch id for video stream_name.

> Pertaining to my earlier comment at [#55 (comment)](https://github.com/aws-robotics/aws-robomaker-sample-application-persondetection/pull/55#discussion_r333242010), not sure if this condition is working properly.

I have taken reference from [monitoring.launch.py](https://github.com/aws-robotics/aws-robomaker-sample-application-cloudwatch/blob/ros2/robot_ws/src/cloudwatch_robot/launch/monitoring.launch.py) of [aws-robomaker-sample-application-cloudwatch](https://github.com/aws-robotics/aws-robomaker-sample-application-cloudwatch/tree/ros2) (can see for log_group_name and aws_metrics_namespace) to make implementation as aws standard and in the same way as other applications.

I have implemented the same as cloudwatch application as you can see that below.
[@L39](https://github.com/aws-robotics/aws-robomaker-sample-application-persondetection/blob/6be135bfd951c9e71e97f43b407e246562484d81/robot_ws/src/person_detection_robot/launch/kinesis.launch.py#L39), getting stream name from the config 

`    default_stream_name = config_yaml['kinesis_video_streamer']['ros__parameters']['kinesis_video']['stream0']['stream_name']`

[@L47](https://github.com/aws-robotics/aws-robomaker-sample-application-persondetection/blob/6be135bfd951c9e71e97f43b407e246562484d81/robot_ws/src/person_detection_robot/launch/kinesis.launch.py#L47) declared as an argument 
```
        launch.actions.DeclareLaunchArgument(
            name='launch_id',
            description='Used for resource name suffix if specified',
            default_value=launch.substitutions.EnvironmentVariable('LAUNCH_ID')
        ),
```
set as launch configuration [@L63](https://github.com/aws-robotics/aws-robomaker-sample-application-persondetection/blob/6be135bfd951c9e71e97f43b407e246562484d81/robot_ws/src/person_detection_robot/launch/kinesis.launch.py#L63). 
```
        launch.actions.SetLaunchConfiguration(
            name='stream_name',
            value=PythonExpression(["'", LaunchConfiguration('stream_name'), "-", LaunchConfiguration('launch_id'), "'"]),
            condition=IfCondition(PythonExpression(["'true' if '", LaunchConfiguration('launch_id'), "' else 'false'"]))
        ),
```
And [@L86](https://github.com/aws-robotics/aws-robomaker-sample-application-persondetection/blob/6be135bfd951c9e71e97f43b407e246562484d81/robot_ws/src/person_detection_robot/launch/kinesis.launch.py#L86), I have taken stream_name as an argument but kinesis.launch.py is not accepting any argument except config and node_node. I have already conveyed the issue over mail (regarding LAUNCH ID argument).
```
        launch.actions.IncludeLaunchDescription(
            launch.launch_description_sources.PythonLaunchDescriptionSource(
                os.path.join(get_package_share_directory('kinesis_video_streamer'), 'launch', 'kinesis_video_streamer.launch.py')
            ),
            launch_arguments={
                'node_name': launch.substitutions.LaunchConfiguration('kinesis_node_name'),
                'config': kvs_config_file_path, 
                'log4cplus_config': kvs_logger_config_file_path,
                'stream_name': launch.substitutions.LaunchConfiguration('stream_name'),
            }.items()
        )
```

Reply:
Okay, let's not block on this right now.

Reply:
Ok, For launch_id issue, I will test once I will get input from AAlon and will do changes if needed.